From f8db95955349b1b372198978c03db104b0c05ff3 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Tue, 20 Feb 2024 11:04:39 +0530
Subject: [PATCH 113/N] CI: clean up ignored tests

Remove tests which are passing now, or no longer exist, and
move tests which only fail with msvcrt into a separate file.
---
 .github/workflows/mingw.yml | 12 ++++++++++--
 Lib/test/test_wmi.py        |  2 +-
 mingw_ignorefile.txt        | 24 +++---------------------
 mingw_ignorefile_msvcrt.txt |  8 ++++++++
 4 files changed, 22 insertions(+), 24 deletions(-)
 create mode 100644 mingw_ignorefile_msvcrt.txt

diff --git a/.github/workflows/mingw.yml b/.github/workflows/mingw.yml
index 0c9f644..827dfe5 100644
--- a/.github/workflows/mingw.yml
+++ b/.github/workflows/mingw.yml
@@ -89,16 +89,24 @@ jobs:
         shell: msys2 {0}
         run: |
           IGNOREFILE="$(pwd)/mingw_ignorefile.txt"
+          IGNOREFILE_EXTRA="$IGNOREFILE"
+          if [[ "${{ matrix.msystem }}" == "MINGW32" ]] || [[ "${{ matrix.msystem }}" == "MINGW64" ]]; then
+            IGNOREFILE_EXTRA="$(pwd)/mingw_ignorefile_msvcrt.txt"
+          fi
           cd _build
-          MSYSTEM= ./python.exe -m test -j8 --ignorefile "$IGNOREFILE" -W
+          MSYSTEM= ./python.exe -m test -j8 --ignorefile "$IGNOREFILE" --ignorefile "$IGNOREFILE_EXTRA" -W
 
       - name: Run broken tests
         continue-on-error: true
         shell: msys2 {0}
         run: |
           IGNOREFILE="$(pwd)/mingw_ignorefile.txt"
+          IGNOREFILE_EXTRA="$IGNOREFILE"
+          if [[ "${{ matrix.msystem }}" == "MINGW32" ]] || [[ "${{ matrix.msystem }}" == "MINGW64" ]]; then
+            IGNOREFILE_EXTRA="$(pwd)/mingw_ignorefile_msvcrt.txt"
+          fi
           cd _build
-          MSYSTEM= ./python.exe -m test -j8 --matchfile "$IGNOREFILE" -W
+          MSYSTEM= ./python.exe -m test -j8 --matchfile "$IGNOREFILE" --matchfile "$IGNOREFILE_EXTRA" -W
 
       - name: Install
         shell: msys2 {0}
diff --git a/Lib/test/test_wmi.py b/Lib/test/test_wmi.py
index f667926..f1490a3 100644
--- a/Lib/test/test_wmi.py
+++ b/Lib/test/test_wmi.py
@@ -7,7 +7,7 @@ from test.support import import_helper, requires_resource, LOOPBACK_TIMEOUT
 
 
 # Do this first so test will be skipped if module doesn't exist
-_wmi = import_helper.import_module('_wmi', required_on=['win'])
+_wmi = import_helper.import_module('_wmi')
 
 
 def wmi_exec_query(query):
diff --git a/mingw_ignorefile.txt b/mingw_ignorefile.txt
index e692d7f..550e724 100644
--- a/mingw_ignorefile.txt
+++ b/mingw_ignorefile.txt
@@ -2,16 +2,11 @@ test.test_ctypes.test_loading.LoaderTest.test_load_dll_with_flags
 distutils.tests.test_bdist_dumb.BuildDumbTestCase.test_simple_built
 distutils.tests.test_cygwinccompiler.CygwinCCompilerTestCase.test_get_versions
 distutils.tests.test_util.UtilTestCase.test_change_root
-test.datetimetester.TestLocalTimeDisambiguation_Fast.*
-test.datetimetester.TestLocalTimeDisambiguation_Pure.*
-test.test_cmath.CMathTests.test_specific_values
-test.test_cmd_line_script.CmdLineTest.test_consistent_sys_path_for_direct_execution
 test.test_compileall.CommandLineTestsNoSourceEpoch.*
 test.test_compileall.CommandLineTestsWithSourceEpoch.*
 test.test_compileall.CompileallTestsWithoutSourceEpoch.*
 test.test_compileall.CompileallTestsWithSourceEpoch.*
 test.test_import.ImportTests.test_dll_dependency_import
-test.test_math.MathTests.*
 test.test_ntpath.NtCommonTest.test_import
 test.test_os.StatAttributeTests.test_stat_block_device
 test.test_os.TestScandir.test_attributes
@@ -23,21 +18,8 @@ test.test_site._pthFileTests.*
 test.test_site.HelperFunctionsTests.*
 test.test_site.StartupImportTests.*
 test.test_ssl.*
-test.test_strptime.CalculationTests.*
-test.test_strptime.StrptimeTests.test_weekday
-test.test_strptime.TimeRETests.test_compile
-test.test_tools.test_i18n.Test_pygettext.test_POT_Creation_Date
 test.test_venv.BasicTest.*
 test.test_venv.EnsurePipTest.*
-test.test_sysconfig.TestSysConfig.test_user_similar
-test.test_tcl.TclTest.testLoadWithUNC
-test.test_wmi
-# flaky
-test.test__xxsubinterpreters.*
-test.test_asyncio.test_subprocess.SubprocessProactorTests.test_stdin_broken_pipe
-test.test_asynchat.TestAsynchat.test_line_terminator2
-test.test_asyncgen.AsyncGenAsyncioTest.test_async_gen_asyncio_gc_aclose_09
-test.test_concurrent_futures.ThreadPoolShutdownTest.test_interpreter_shutdown
-test.test_asynchat.TestNotConnected.test_disallow_negative_terminator
-test.test_logging.SysLogHandlerTest.*
-test.test_logging.IPv6SysLogHandlerTest.*
+
+test.test_dict.DictTest.test_splittable_to_generic_combinedtable
+test.test_capi.test_run.CAPITest.test_run_fileexflags
diff --git a/mingw_ignorefile_msvcrt.txt b/mingw_ignorefile_msvcrt.txt
new file mode 100644
index 0000000..be363bd
--- /dev/null
+++ b/mingw_ignorefile_msvcrt.txt
@@ -0,0 +1,8 @@
+test.datetimetester.TestLocalTimeDisambiguation_Fast.*
+test.datetimetester.TestLocalTimeDisambiguation_Pure.*
+test.test_cmath.CMathTests.test_specific_values
+test.test_math.MathTests.*
+test.test_strptime.CalculationTests.*
+test.test_strptime.StrptimeTests.test_weekday
+test.test_strptime.TimeRETests.test_compile
+test.test_tools.test_i18n.Test_pygettext.test_POT_Creation_Date
